### PR TITLE
Audio volume and spoopy music

### PR DESCRIPTION
**Pull request by NiceGuyAndersson**
SoundManager now takes individual sources' volume into account.
Also added some spoopy music.

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 8 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ScriptManager.cpp#L654](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/961285cd0696e6d9ceb2642f1e9c54a98d079f11/src/Engine/Manager/ScriptManager.cpp#L654): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L658](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/961285cd0696e6d9ceb2642f1e9c54a98d079f11/src/Engine/Manager/ScriptManager.cpp#L658): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L662](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/961285cd0696e6d9ceb2642f1e9c54a98d079f11/src/Engine/Manager/ScriptManager.cpp#L662): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L676](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/961285cd0696e6d9ceb2642f1e9c54a98d079f11/src/Engine/Manager/ScriptManager.cpp#L676): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L684](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/961285cd0696e6d9ceb2642f1e9c54a98d079f11/src/Engine/Manager/ScriptManager.cpp#L684): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [SoundManager.cpp#L158](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/961285cd0696e6d9ceb2642f1e9c54a98d079f11/src/Engine/Manager/SoundManager.cpp#L158): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![INFO][INFO] [EntityEditor.cpp#L443](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/961285cd0696e6d9ceb2642f1e9c54a98d079f11/src/Editor/GUI/Editors/EntityEditor.cpp#L443): keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 8 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ScriptManager.cpp#L654](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f3c496832970e72722630d8695ab03a9c988ba82/src/Engine/Manager/ScriptManager.cpp#L654): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L658](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f3c496832970e72722630d8695ab03a9c988ba82/src/Engine/Manager/ScriptManager.cpp#L658): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L662](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f3c496832970e72722630d8695ab03a9c988ba82/src/Engine/Manager/ScriptManager.cpp#L662): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L676](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f3c496832970e72722630d8695ab03a9c988ba82/src/Engine/Manager/ScriptManager.cpp#L676): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L684](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f3c496832970e72722630d8695ab03a9c988ba82/src/Engine/Manager/ScriptManager.cpp#L684): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [SoundManager.cpp#L158](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f3c496832970e72722630d8695ab03a9c988ba82/src/Engine/Manager/SoundManager.cpp#L158): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![INFO][INFO] [EntityEditor.cpp#L443](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/f3c496832970e72722630d8695ab03a9c988ba82/src/Editor/GUI/Editors/EntityEditor.cpp#L443): keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 8 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ScriptManager.cpp#L654](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c55f93dc921e960d8771fc26243d740efdb9e6f0/src/Engine/Manager/ScriptManager.cpp#L654): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L658](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c55f93dc921e960d8771fc26243d740efdb9e6f0/src/Engine/Manager/ScriptManager.cpp#L658): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L662](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c55f93dc921e960d8771fc26243d740efdb9e6f0/src/Engine/Manager/ScriptManager.cpp#L662): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L676](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c55f93dc921e960d8771fc26243d740efdb9e6f0/src/Engine/Manager/ScriptManager.cpp#L676): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L684](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c55f93dc921e960d8771fc26243d740efdb9e6f0/src/Engine/Manager/ScriptManager.cpp#L684): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [SoundManager.cpp#L158](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c55f93dc921e960d8771fc26243d740efdb9e6f0/src/Engine/Manager/SoundManager.cpp#L158): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![INFO][INFO] [EntityEditor.cpp#L443](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c55f93dc921e960d8771fc26243d740efdb9e6f0/src/Editor/GUI/Editors/EntityEditor.cpp#L443): keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 8 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ScriptManager.cpp#L659](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/dfeacd7ea3bb79bdecf2c5f83653eed34b8f1548/src/Engine/Manager/ScriptManager.cpp#L659): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L663](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/dfeacd7ea3bb79bdecf2c5f83653eed34b8f1548/src/Engine/Manager/ScriptManager.cpp#L663): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L667](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/dfeacd7ea3bb79bdecf2c5f83653eed34b8f1548/src/Engine/Manager/ScriptManager.cpp#L667): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L681](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/dfeacd7ea3bb79bdecf2c5f83653eed34b8f1548/src/Engine/Manager/ScriptManager.cpp#L681): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L689](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/dfeacd7ea3bb79bdecf2c5f83653eed34b8f1548/src/Engine/Manager/ScriptManager.cpp#L689): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [SoundManager.cpp#L158](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/dfeacd7ea3bb79bdecf2c5f83653eed34b8f1548/src/Engine/Manager/SoundManager.cpp#L158): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![INFO][INFO] [EntityEditor.cpp#L443](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/dfeacd7ea3bb79bdecf2c5f83653eed34b8f1548/src/Editor/GUI/Editors/EntityEditor.cpp#L443): keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Component/SoundSource.hpp [hunk 1]
```diff
@@ -39,11 +39,8 @@ namespace Component {
             /// Sound buffer.
             Audio::SoundBuffer* soundBuffer = nullptr;
             
-            /// Pitch.
-            float pitch = 1.f;
-            
             /// Gain.
```
Comment (andens) [inline src/Engine/Component/SoundSource.hpp]: This is volume now.

File: Game/Resources/Opening Scene/Opening Scene.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
 			"Material" : 
 			{
 				"albedo" : "Resources/Opening Scene/Cave/Textures/cave_albedo",
-				"metallic" : "Resources/Opening Scene/Cave/Textures/cave_metallic",
+				"metallic" : "Resources/Opening Scene/Cave/Textures/Ambient",
```
Comment (andens) [inline Game/Resources/Opening Scene/Opening Scene.json]: Any particular reason why this texture was changed to something called 'Ambient'?

Comment (NiceGuyAndersson) [reply Game/Resources/Opening Scene/Opening Scene.json]: Must've been a weird bug when importing the audio. Will try to reproduce and write an issue.

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 8 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ScriptManager.cpp#L659](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8d6ae4a1ebbfd86fc7b00d35daa3a9fb4c964bb8/src/Engine/Manager/ScriptManager.cpp#L659): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L663](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8d6ae4a1ebbfd86fc7b00d35daa3a9fb4c964bb8/src/Engine/Manager/ScriptManager.cpp#L663): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L667](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8d6ae4a1ebbfd86fc7b00d35daa3a9fb4c964bb8/src/Engine/Manager/ScriptManager.cpp#L667): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L681](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8d6ae4a1ebbfd86fc7b00d35daa3a9fb4c964bb8/src/Engine/Manager/ScriptManager.cpp#L681): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L689](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8d6ae4a1ebbfd86fc7b00d35daa3a9fb4c964bb8/src/Engine/Manager/ScriptManager.cpp#L689): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [SoundManager.cpp#L159](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8d6ae4a1ebbfd86fc7b00d35daa3a9fb4c964bb8/src/Engine/Manager/SoundManager.cpp#L159): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![INFO][INFO] [EntityEditor.cpp#L443](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/8d6ae4a1ebbfd86fc7b00d35daa3a9fb4c964bb8/src/Editor/GUI/Editors/EntityEditor.cpp#L443): keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 8 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ScriptManager.cpp#L659](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ec4b1f46d71bfb832868a90d2d7f00c29236da76/src/Engine/Manager/ScriptManager.cpp#L659): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L663](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ec4b1f46d71bfb832868a90d2d7f00c29236da76/src/Engine/Manager/ScriptManager.cpp#L663): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MAJOR][MAJOR] [ScriptManager.cpp#L667](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ec4b1f46d71bfb832868a90d2d7f00c29236da76/src/Engine/Manager/ScriptManager.cpp#L667): Memory leak: value [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3Amemleak)
1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ScriptManager.cpp#L681](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ec4b1f46d71bfb832868a90d2d7f00c29236da76/src/Engine/Manager/ScriptManager.cpp#L681): The function 'GetScripts' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [ScriptManager.cpp#L689](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ec4b1f46d71bfb832868a90d2d7f00c29236da76/src/Engine/Manager/ScriptManager.cpp#L689): The function 'ExecuteScriptMethod' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [SoundManager.cpp#L159](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ec4b1f46d71bfb832868a90d2d7f00c29236da76/src/Engine/Manager/SoundManager.cpp#L159): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![INFO][INFO] [EntityEditor.cpp#L443](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ec4b1f46d71bfb832868a90d2d7f00c29236da76/src/Editor/GUI/Editors/EntityEditor.cpp#L443): keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**